app/portainer/components/code-editor/codeEditorController.js

```diff
@@ -1,6 +1,15 @@
 angular.module('portainer.app').controller('CodeEditorController', function CodeEditorController($document, CodeMirrorService, $scope) {
   var ctrl = this;
 
+  $scope.$watch(
+    () => ctrl.readOnly,
+    () => {
+      if (ctrl.editor && ctrl.editor.getValue('readOnly') !== ctrl.readOnly) {
+        ctrl.editor.setOption('readOnly', ctrl.readOnly);
+      }
+    }
+  );
```

Comment:
why not use `$onChanges`?

Reply:
Maybe I was doing it wrong, but it was not triggering on changes to the `readOnly` binding

Reply:
Do you have an example of what it should look like?